quickwit-indexing/src/actors/merge_executor.rs

```diff
@@ -231,10 +251,75 @@ fn merge_split_directories(
     Ok(output_directory)
 }
 
+/// Applies delete tasks on the directory stack, commit, and merge to rebuild the split
+/// without the deleted documents.
+/// The directory stack is made of:
+/// - The output directory that is a `MmapDirectory` on `output_path`. This directory receives all
+///   writes.
+/// - The current split directory which is expected to be a `BundleDirectory` as the split has just
+///   been downloaded. This `BundleDirectory` is read-only.
+fn apply_deletes_and_merge_split(
```

Comment:
Could this be the same function for merges? (With an empty vec for delete_tasks)

Reply:
Indeed, done.